### ISSUE TITLE
plot_source ignores the rate setting of a dataset

### ISSUE THREAD

**Issue by DougBurke** [sherpa/sherpa#1087]
```
get_data().rate = True
plot('source', 'model')
get_data().rate = False
plot('source', 'model')
```

creates 

![sherpa](https://user-images.githubusercontent.com/224638/106279281-297a2d80-620a-11eb-97b3-371811fa6402.png)

and then

![sherpa](https://user-images.githubusercontent.com/224638/106279325-3c8cfd80-620a-11eb-8a37-30fc0bc51e0f.png)

Note that the source plot has not changed y units. Changing get_data().plot_fac does change the source plot.

Is this something we care about or not?

**Comment by hamogu** [comment]
I think we do care.

**Comment by hamogu** [comment]
And is it not the best to stumble on questions like this while celebrating a major birthday?

**Comment by DougBurke** [comment]
Unfortunately I've had this in my back pocket for a day or two. I just like to space out my github profile...

**Comment by DougBurke** [comment]
So, let's consider the case of the model `xspowerlaw.pl`. When we call `plot_source` we evaluate the model `pl`, and when we evaluate `plot_model` we evaluate the model `rmf(arf(exposure * pl))` (for the case of a "standard" PHA case where we are not doing anything fancy). For the `plot_model` case we have an exposure time which we can include or not, whereas for the `plot_source` case we don't. This is important because the models (in this case `pl`) are taken to return a value per-second (or, can be thought of as representing a single-second exposure), which is why the `exposure` term gets automatically added for `plot_model`. 

Therefore, at some level setting `set_analysis(..., type='counts')` or `get_data().rate = False` could be considered to be an error: we don't have an exposure time to use. Now, technically we do have an exposure time (from the data object), so we could use it for the `counts` case, but then we'd be plotting `photons/cm^2/keV` (or, with `plot_fac=1`, `photons/cm^2`), and I'm not sure what use this has (other than "doing something different" when the rate setting is changed).

**Comment by hamogu** [comment]
I have a nagging thought that tells me that we are rapidly nearing the end of where we can get with Sherpa's simplistic units system. Astropy's units system is build to deal exactly with cases like this and it's far more complex. I'm sure we can fix this one thing, but then how about nm as unit of wavelength or frequency?  My understanding is that tools like [3ML](https://threeml.readthedocs.io/en/stable/index.html) use astropy.units for this exact reason.

**Comment by DougBurke** [comment]
Closed by #1828
